cajita/src/Cajita_FastFourierTransform.hpp

```diff
@@ -214,21 +216,35 @@ class FastFourierTransform
             global_low[Dim::K] + local_num_entity[Dim::K] - 1};
 
         // Setup the fft.
-        int permute = 0;
-        int fftsize, sendsize, recvsize;
-        _fft.setup( global_num_entity.data(), global_low.data(),
-                    global_high.data(), global_low.data(), global_high.data(),
-                    permute, fftsize, sendsize, recvsize );
+        //! int permute = 0;
+        //! int fftsize, sendsize, recvsize; 
+        int fftsize;
+
+        //! Old heFFTe plan creation
+        // _fft.setup( global_num_entity.data(), global_low.data(),
+        //             global_high.data(), global_low.data(), global_high.data(),
+        //             permute, fftsize, sendsize, recvsize );
+
+        heffte::fft3d<Backend> fft( heffte::box3d( global_low.data(), global_high.data() ) ,
+                                        heffte::box3d( global_low.data(), global_high.data() ) ,
+                                        layout.localGrid()->globalGrid().comm(), params );
+
+        fftsize = std::max(fft.size_outbox(), fft.size_inbox());
 
         // Check the size.
         if ( fftsize < (int)entity_space.size() )
-            throw std::logic_error( "HEFFTE expected allocation size smaller "
+            throw std::logic_error( "heFFTe expected allocation size smaller "
                                     "than local grid size" );
 
         // Allocate the work array.
         _fft_work = Kokkos::View<Scalar *, DeviceType>(
             Kokkos::ViewAllocateWithoutInitializing( "fft_work" ),
-            2 * fftsize );
+            2 * fftsize ); 
```

Comment:
I'm somewhat confused here...  What I understand is you could use just allocate `fftsize` if defining input and output data to be complex? This would be for a complex-to-complex, in-place FFT. 

Could we do something like:
 ```
 _fft_work = Kokkos::View<Kokkos::complex<Scalar> *, DeviceType>(
            Kokkos::ViewAllocateWithoutInitializing( "fft_work" ),
            fftsize );

```